### ISSUE TITLE
is there a multigpu example option?

### ISSUE THREAD

**Issue by danieltanhx** [Shilin-LU/TF-ICON#6]

**Comment by Shilin-LU** [comment]
Thank you for reaching out with your question. Currently, we do not have a multi-GPU option since we do not involve a training phase.

**Comment by 1siddhi7** [comment]
Hi! Just had a comment in case its helpful!
I was running this codebase on a GPU cluster and got "RuntimeError: CUDA error: out of memory."

Making one change in line 300 of main_tf_icon.py fixed it:

`with torch.cuda.device(opt.gpu):`
      `torch.cuda.empty_cache()`

**Comment by Shilin-LU** [comment]
> Hi! Just had a comment in case its helpful! I was running this codebase on a GPU cluster and got "RuntimeError: CUDA error: out of memory."
> 
> Making one change in line 300 of main_tf_icon.py fixed it:
> 
> `with torch.cuda.device(opt.gpu):` `torch.cuda.empty_cache()`

Thanks for your helpful comments!

**Comment by Shilin-LU** [comment]
@danieltanhx  Hi, I've identified a simple way to reduce memory usage in the latest version. One effective solution is to call `torch.cuda.empty_cache()` more frequently, as pointed out in this [section of the code](https://github.com/Shilin-LU/TF-ICON/blob/163cd67fb9605173231210ab16486ee86f269aff/ptp_scripts/ptp_utils.py#L192).

During our evaluation, we observed that the memory consumption was limited to a maximum of 22 GB. Hope this helps!

I encourage you to clone the updated version of our repository and give it another shot.
